### PR TITLE
Do not add sensitive information in Ceph log files

### PR DESCRIPTION
**Pull request by neha-ojha**
Over time, the "config set" and "config-key set" commands have been used to set configuration keys whose values hold sensitive data like passwords. These commands(including the passwords) are logged at every level in audit, mon, mgr, cluster logs in plaintext, which poses a security threat. This PR avoids logging passwords in Ceph logs by not logging values for  the "config set" and "config-key set" commands in some places and not logging the entire command in other places. Some debug statements have been left out as comments for development ease but users are expected to use "ceph config log" and "ceph config dump" to check configuration values. In future, after we identify the exact set of configuration options that are sensitive, we can avoid logging only those, but until then, we need a blanket solution to address this security issue.

Fixes: https://tracker.ceph.com/issues/37503
Signed-off-by: Neha Ojha <nojha@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by neha-ojha** [comment]
https://pulpito.ceph.com/nojha-2020-12-06_00:38:18-rados-wip-37503-3-distro-basic-smithi/ - no related failures

File: src/messages/MMonCommand.h [hunk 1]
```diff
@@ -41,10 +42,26 @@ class MMonCommand final : public PaxosServiceMessage {
 public:
   std::string_view get_type_name() const override { return "mon_command"; }
   void print(std::ostream& o) const override {
+    cmdmap_t cmdmap;
```
Comment (tchaikov) [inline src/messages/MMonCommand.h:47]: please move the implementation into .cc.

**Review by tchaikov — DISMISSED** [review]

File: src/messages/MMonCommand.h (hunk 1 above)
Comment (neha-ojha) [reply src/messages/MMonCommand.h:47]: there is currently no .cc file in https://github.com/ceph/ceph/tree/master/src/messages and for the simplicity of backport until nautilus, can we agree to leave it here?

Comment (tchaikov) [reply src/messages/MMonCommand.h:47]: fair enough. will create a follow-up PR to address this.

File: src/messages/MMonCommandAck.h [hunk 2]
```diff
@@ -33,7 +34,27 @@ class MMonCommandAck final : public PaxosServiceMessage {
...
+      ceph::common::cmd_getval(cmdmap, "name", name);
+      o << "[{prefix=" << prefix
+        << ", name=" << name << "}]"
+        << "=" << r << " " << rs << " v" << version << ")";
+    } else if (prefix == "config-key set") {
+      string key;
+      ceph::common::cmd_getval(cmdmap, "key", key);
+      o << "[{prefix=" << prefix << ", key=" << key << "}]"
+        << "=" << r << " " << rs << " v" << version << ")";
+    } else
```
Comment (jdurgin) [inline src/messages/MMonCommandAck.h:57]: nit: else clause should be surrounded by {} since the first block of the if used them

**Review by jdurgin — APPROVED** [review]

**Review by epuertat — APPROVED** [review]
Looks good, @neha-ojha ! Thank you.

As I don't usually set debug log levels, I didn't realize the ceph-mgr was logging values in these other locations!

File: src/messages/MMonCommandAck.h (hunk 2 above)
Comment (neha-ojha) [reply src/messages/MMonCommandAck.h]: sure, done

**Comment by neha-ojha** [comment]
@jdurgin I see you added needs-qa, this PR has already been tested https://github.com/ceph/ceph/pull/38479#issuecomment-740153552 and the only change after that was to address https://github.com/ceph/ceph/pull/38479#discussion_r542996164. ok to merge?

**Comment by jdurgin** [comment]
i missed that comment, thanks!

**Pull request merged**